ci-operator/step-registry/gather/alibabacloud-console/gather-alibabacloud-console-commands.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+fi
+
+if test -f "${SHARED_DIR}/alibaba-instance-ids.txt"
+then
+
+	export ALIBABA_CLOUD_CREDENTIALS_FILE="${SHARED_DIR}/alibabacreds.ini"
+	export ALIBABA_CLI_CREDENTIALS_FILE="${SHARED_DIR}/config"
+
+	pushd /tmp
+	wget https://aliyuncli.alicdn.com/aliyun-cli-linux-latest-amd64.tgz -O aliyun-cli.tgz
+	tar zxvf aliyun-cli.tgz
```

Comment:
Not sure how worried you are about running this freshly downloaded executable.  There are certainly many other steps that do similar stuff.  But if you wanted to pin a specific version and confirm that the binary you received had not been corrupted, [this][1] is an example of one way to check.

[1]: https://github.com/openshift/release/blob/b5f02576fc1df4eba8d6521bb3c600868a9f5733/ci-operator/step-registry/openshift/manifests/external/openshift-manifests-external-commands.sh#L7-L13

Reply:
This was spawned from our previous discussion. I filed an issue and it will be addressed https://github.com/aliyun/aliyun-cli/issues/401